src/github.com/getlantern/cloudflare/record.go

```diff
@@ -0,0 +1,351 @@
+	ServiceMode string
+}
+
+// UpdateRecord destroys a record by the ID specified and
+// returns an error if it fails. If no error is returned,
+// the Record was succesfully updated.
+func (c *Client) UpdateRecord(domain string, id string, opts *UpdateRecord) error {
+	params := make(map[string]string)
+	params["z"] = domain
+	params["id"] = id
+
+	params["type"] = opts.Type
+
+	if opts.Name != "" {
+		params["name"] = opts.Name
+	}
+
+	if opts.Content != "" {
+		params["content"] = opts.Content
+	}
+
+	if opts.Priority != "" {
+		params["prio"] = opts.Priority
+	}
+
+	if opts.Ttl != "" {
+		params["ttl"] = opts.Ttl
+	}
+
+	if opts.ServiceMode != "" {
+		params["service_mode"] = opts.ServiceMode
+	}
+
+	req, err := c.NewRequest(params, "POST", "rec_edit")
+	if err != nil {
+		return err
+	}
+
+	resp, err := checkResp(c.Http.Do(req))
+	if resp != nil && resp.Body != nil {
+		defer resp.Body.Close()
+	}
+	if err != nil {
+		return fmt.Errorf("Error updating record: %s", err)
+	}
+
+	recordResp := new(RecordResponse)
+
+	err = decodeBody(resp, &recordResp)
+
+	if err != nil {
+		return fmt.Errorf("Error parsing record response: %s", err)
+	}
+	_, err = recordResp.GetRecord()
+	if err != nil {
+		return err
+	}
+
+	// The request was successful
+	return nil
+}
+
+// RetrieveRecord gets  a record by the ID specified and
+// returns a Record and an error. An error will be returned for failed
+// requests with a nil Record.
+func (c *Client) RetrieveRecord(domain string, id string) (*Record, error) {
+	records, err := c.LoadAll(domain)
+	if err != nil {
+		return nil, err
+	}
+
+	record, err := records.FindRecord(id)
+	if err != nil {
+		return nil, err
+	}
+
+	// The request was successful
+	return record, nil
+}
+
+// RetrieveRecord gets  a record by the ID specified and
+// returns a Record and an error. An error will be returned for failed
+// requests with a nil Record.
+func (c *Client) RetrieveRecordByName(domain string, name string) (*Record, error) {
+	records, err := c.LoadAll(domain)
+	if err != nil {
+		return nil, err
+	}
+
+	record, err := records.FindRecordByName(name)
+	if err != nil {
+		return nil, err
+	}
+
+	// The request was successful
+	return record, nil
+}
+
+func (c *Client) LoadAll(domain string) (*RecordsResponse, error) {
+	params := make(map[string]string)
+	// The zone we want
+	params["z"] = domain
+	return c.loadAll(&params)
+}
+
+func (c *Client) LoadAllAtIndex(domain string, index int) (*RecordsResponse, error) {
+	params := make(map[string]string)
+	// The zone we want
+	params["z"] = domain
+	params["o"] = strconv.Itoa(index)
+	return c.loadAll(&params)
+}
+
+func (c *Client) loadAll(params *map[string]string) (*RecordsResponse, error) {
+	req, err := c.NewRequest(*params, "GET", "rec_load_all")
+
+	if err != nil {
+		return nil, fmt.Errorf("Error creating request: %s", err)
+	}
+
+	resp, err := checkResp(c.Http.Do(req))
+	if resp != nil && resp.Body != nil {
+		defer func() {
+			resp.Body.Close()
```

Comment:
Using defer like this could lead to race conditions, actually not in this case, unless we set `resp.Body` to `nil` before returning. @oxtoacart I think I could fix small and easy things like this after commenting the code, it totally won't be a problem, should I?